runtime/signals_nat.c

```diff
@@ -230,6 +230,12 @@ DECLARE_SIGNAL_HANDLER(segv_handler)
     Caml_state->young_ptr = (value *) CONTEXT_YOUNG_PTR;
 #endif
     caml_raise_stack_overflow();
+#endif
+  } else if (Caml_state->checking_pointer_pc) {
+#ifdef CONTEXT_PC
+    CONTEXT_PC = (context_reg)Caml_state->checking_pointer_pc;
+#else
+#error "CONTEXT_PC must be defined if RETURN_AFTER_STACK_OVERFLOW is"
```

Comment:
This file is such an `#ifdef` soup already (!!!), but I think this should be guarded with the `NAKED_POINTERS_CHECKER` `#ifdef` (it'll be compulsory if the field isn't there).